torchmetrics/utilities/data.py

```diff
@@ -238,3 +238,11 @@ def get_group_indexes(indexes: Tensor) -> List[Tensor]:
             res[_id] = [i]
 
     return [tensor(x, dtype=torch.long) for x in res.values()]
+
+
+def _squeeze_scalar_element_tensor(x: Tensor) -> Tensor:
+    return x.squeeze() if x.numel() == 1 else x
+
+
+def _squeeze_if_scalar(data: Any) -> Any:
+    return apply_to_collection(data, Tensor, _squeeze_scalar_element_tensor)
```

Comment:
noob q: why don't we update as many of the following to use `torch.tensor(0)`

https://grep.app/search?q=default%3Dtorch.zeros%28&filter[repo][0]=PyTorchLightning/metrics

Reply:
@SkafteNicki ^^

Reply:
Updated now. Still believe that a global squeeze is good to have to secure that we not in the future mess up.